pkg/es/writer/esBulkHandler.go

```diff
@@ -369,44 +370,35 @@ func AddAndGetRealIndexName(indexNameIn string, localIndexMap map[string]string,
 	return indexNameConverted
 }
 
-func ProcessIndexRequest(rawJson []byte, tsNow uint64, indexNameIn string,
-	bytesReceived uint64, flush bool, localIndexMap map[string]string, myid uint64,
-	rid uint64, idxToStreamIdCache map[string]string,
-	cnameCacheByteHashToStr map[uint64]string, jsParsingStackbuf []byte) error {
-	indexNameConverted := AddAndGetRealIndexName(indexNameIn, localIndexMap, myid)
-	cfgkey := config.GetTimeStampKey()
-
-	var docType segment.SIGNAL_TYPE
-	if strings.HasPrefix(indexNameConverted, "jaeger-") {
-		docType = segment.SIGNAL_JAEGER_TRACES
-		cfgkey = "startTimeMillis"
-	} else {
-		docType = segment.SIGNAL_EVENTS
-	}
-
-	tsMillis := utils.ExtractTimeStamp(rawJson, &cfgkey)
+func GetNewPLE(rawJson []byte, tsNow uint64, indexName string, tsKey *string, jsParsingStackbuf []byte) (*writer.ParsedLogEvent, error) {
```

Comment:
nit: it might make more sense to move this helper function into the same package where the PLE type is defined